backend/src/channel/channel.service.ts

```diff
@@ -0,0 +1,512 @@
+                channel_id: channel_id,
+                user_id: user.id,
+            },
+        });
+        if (!channelMember) {
+            return 'User is not a member of the channel';
+        }
+        if (channelMember.MemberType !== 'OWNER' && channelMember.MemberType !== 'ADMIN') {
+            return 'User is not the owner of the channel';
+        }
+        const updatedChannel = await prisma.channel.update({
+            where: {
+                id: channel_id,
+            },
+            data: {
+                name: data.name,
+                type: data.type,
+                password: data.password,
+                description: data.description,
+            },
+        });
+        return {
+            id: updatedChannel.id,
+            name: updatedChannel.name,
+            Type: updatedChannel.type,
+            createdAt: updatedChannel.createdAt,
+            updatedAt: updatedChannel.updatedAt,
+        };
+    }
+
+    async deleteChannel(@Req() req: Request, channel_id: number) {
+        channel_id = Number(channel_id);
+        const channel = await prisma.channel.findUnique({
+            where: {
+                id: channel_id,
+            },
+            include: {
+                ChannelMembers: true,
+            },
+        });
+        if (!channel) {
+            return 'Channel not found';
+        }
+        const user = await this.searchUserById(req.user['id']);
+        if (!user) {
+            return 'User not found';
+        }
+        const channelMember = await prisma.channelMembers.findFirst({
+            where: {
+                channel_id: channel_id,
+                user_id: user.id,
+            },
+        });
+        if (!channelMember) {
+            return 'User is not a member of the channel';
+        }
+        if (channelMember.MemberType !== 'OWNER') {
+            return 'User is not the owner of the channel';
+        }
+        await prisma.channel.delete({
+            where: {
+                id: channel_id,
+            },
+        });
+        return 'Channel deleted';
+    }
+
+    async createChannelMember(userId: number, channelId: number, memberType: 'MEMBER' | 'REQUESTED' | 'INVITED') {
+        return await prisma.channelMembers.create({
+            data: {
+                user_id: userId,
+                channel_id: channelId,
+                MemberType: memberType
+            }
+        });
+    }
+
+    async updateChannelMember(memberId: number, memberType: 'MEMBER' | 'NOTMEMBER' | 'BANNED' | 'MUTED' | 'ADMIN') {
+        return await prisma.channelMembers.update({
+            where: {
+                id: memberId
+            },
+            data: {
+                MemberType: memberType
+            }
+        });
+    }
+
+
+    async AdminActions(@Req() req: Request, channel_id: number, user_id: number, action: 'ban' | 'mute' | 'kick' | 'unban' | 'unmute' | 'accept' | 'reject' | 'invite' | 'remove' | 'makeAdmin' | 'makeUser') {

```

Comment:
- you can ban/kick/mute users who rejected the invite or left the channel
- users who are invited can be banned and then unbanned which will force them to be members without accepting the invite
- users who are NOTMEMBER cannot be invited (if the user rejected the invite or left the channel the admin cannot invite him again)
- banned users can be kicked or muted (kicked users can be banned or muted...)
- banned or removed users can be made Admins!